refactor(history): enforce using PK instead of ToxId in history

anthonybilinski opened the pull request:
Need to be leniant when reading from db because of ToxIds being saved in the db
from a bug introduced in e07d8d358f6fc890a77e029aa230b69bdecd325e, which used
self ID rather than self Pk. Was subsequently fixed in
033f28e67ecd61cd0fdb6cc0e8de8a2bc920e1d7.

- [x] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/6130)
<!-- Reviewable:end -->

sphaerophoria reviewed (commented):
Nice, I didn't run this myself, but it looks pretty reasonable. My only concern is switching the type of peers from QHash<QString, int64_t> to QHash<ToxPk, int64_t>, since before we conceivably could have had a peer stored as anything. This seems low risk to me though

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6130#-:-M6RDB-J4tI6utAKmJaK:b-7b7bfw)* status: 0 of 1 LGTMs obtained



<!-- Sent from Reviewable.io -->

sphaerophoria reviewed (commented):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6130#-:-M6RDvFk3TFD2y9l5BjC:b-jbsf8z)* status: :shipit: complete! 1 of 1 LGTMs obtained



<!-- Sent from Reviewable.io -->

anthonybilinski committed:
refactor(history): enforce using PK instead of ToxId in history

We need to be lenient when reading from db because of ToxIds being saved in the
db from a bug introduced in e07d8d358f6fc890a77e029aa230b69bdecd325e, which
used self ID rather than self Pk. Was subsequently fixed in
033f28e67ecd61cd0fdb6cc0e8de8a2bc920e1d7.

anthonybilinski reviewed (commented):
Removed some needless whitespace changes. I think the map change should be fairly safe. ToxPk asserts on size during construction, I've run it on my old profile full of  content, and every construction of ToxPk looks sane (no one just shoving random 32 byte strings into it).

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6130#-:-M6RmxWt0rfQX1sxRmtv:b-dqvq7a)* status: :shipit: complete! 1 of 1 LGTMs obtained



<!-- Sent from Reviewable.io -->
